src/scope-lang/nested-scope-visitor/actions-visitor.ts

```diff
@@ -120,7 +121,8 @@ export class AstVisitor extends PositionVisitor {
 }
 
 // Our visitor has no state, so a single instance is sufficient.
-
+// CR: why is our visitor accepting a **text** input? should not it get the CST directly?
+// Why is it even aware of the lexer module?
 export const toAst = (inputText: string, opts = {}) => {
```

Comment:
This is called by the `onCompletion` handler which only knows the document and document cursor position, and then can fetches the current document being edited and the text of that document and potentially "current cursor context" (text nearby), then call the toAst to generate an AST from that text.

Reply:
I wonder if any implementations cache the parse results of specific texts
for performance reasons, but this optimization probably does not belong in an example scenario.

Reply:
I'm sure they do. I guess you could do a diff and patch or control the parse tree based on that somehow?

Reply:
Lets say you have a "named scope" such as a class or whatever, and no lines within that scope is changed, then reuse that parse tree

Reply:
Yeah, you could even do incremental parsing and modify and existing CST
to a slightly newer CST without re-creating it from scratch (feature does not exist in Chevrotain but should be possible). 

But I guess all these things are out of scope for a simple example. otherwise it becomes too complex to "digest".